src/App/Fossa/Config/ConfigFile.hs

```diff
@@ -257,6 +259,12 @@ newtype ExperimentalGradleConfigs = ExperimentalGradleConfigs
   {gradleConfigsOnly :: Set Text}
   deriving (Eq, Ord, Show)
 
+data MavenScopeConfigs = MavenScopeConfigs
```

Comment:
[Question] Looking at the ticket, it says we want to support EITHER only-scope or exclude-scopes, (similar to how you have it in docs), but this type would mean, user can defined both. 

If we do not want user to define both, we should use Sum Type to restrict possible scenario - it's either include or exclude! 
```
data MavenScopeConfig = MavenScopeOnlyConfig (Set Text) | MavenScopeExcludeConfig (Set Text)
```

Reply:
I spoke to David for clarification on the AC. 

He wrote:
- Hey! I think we can do both only and exclude in the path and target filtering, so it would make sense if we could do both to keep the behavior? In practice, I’d imagine that only one or the other is used
-  i think its 100% fine if theyre mutually exclusive.


With that being I think it makes sense to make it EITHER scope-only or scope-exclude. I don't see the benefit of supporting both and it will declutter the code substantially